backend/db_funcs/cipherGameFuncs.py

```diff
@@ -64,3 +64,35 @@ def getCipherGame(cipher_game_id: int):
     except:
         return {"result": "error"}
     return result
+
+def is_game_organizer(cipher_game_id: int, user_id: int):
+    try:
+        with DB_conn.getConn(connection):
+            with DB_conn.getCursor(connection) as cur:
+                cur.execute("SELECT cipher_game_person.person_id FROM cipher_game_person WHERE cipher_game_person.cipher_game_id = %s;", cipher_game_id)
+                result = cur.fetchall()[0]
+    except:
+        return False
+    return result == user_id
+
+
+def get_visible_ciphers(cipher_game_id: int, user_id: int):
+    try:
+        with DB_conn.getConn(connection):
+            with DB_conn.getCursor(connection) as cur:
+                cur.execute("SELECT cipher.cipher_id FROM cipher WHERE cipher_game_id = %s MINUS SELECT cipher.cipher_id FROM cipher WHERE cipher_game_id = %s AND req_cipher_id NOT IN (SELECT * FROM cipher JOIN attempt ON cipher.cipher_game_id = %s AND attempt.cipher_id = cipher.cipher_id AND attempt.is_successful = TRUE JOIN team ON team.team_id = attempt.team_id JOIN person_team ON person_team.team_id = team.team_id AND person_team.person_id = %s));", (cipher_game_id, cipher_game_id, cipher_game_id , cipher_game_id, user_id,))
+                result = cur.fetchall()
+    except:
+        return None
+    return result
+
+def get_all_ciphers(cipher_game_id: int):
+    try:
+        with DB_conn.getConn(connection):
+            with DB_conn.getCursor(connection) as cur:
+                cur.execute("SELECT cipher.cipher_id FROM cipher WHERE cipher.cipher_game_id = %s;", cipher_game_id)
+                result = cur.fetchall()
+    except:
```

Comment:
Chtělo by to nevracet řádky v databázi, ale samostatné objekty. Zase, copypastnout to, jak to dělám já u uživatelů. Kromě toho bych prosil dopsat `strip` funkci pro šifry, a před vracením z libovolného endpointu ji zavolat - a vymazat informace, které uživatel nepotřebuje znát. Občas tam jsou věci, které by nám uškodily